up_client_socket/rust/u_transport_socket.rs

```diff
@@ -0,0 +1,395 @@
+        let mut socket = &self.socket_sync;
+
+        socket.read(buffer)
+    }
+
+     fn _handle_publish_message(&mut self, umsg: UMessage) {
+        dbg!("HANDLING PUB MESG");
+        // Create a new Tokio runtime
+        let rt = Runtime::new().unwrap();
+   
+        dbg!(&umsg.attributes.source.to_string());
+    
+        if let Some(listner_array) = self
+            .listner_map
+            .lock()
+            .unwrap()
+            .get(&umsg.attributes.source.to_string())
+        {
+            for listner_ref in listner_array {
+                rt.block_on(async {
+                    let _ = listner_ref.on_receive(umsg.clone()).await;
+                });
+            }
+        }
+    }
+
+     fn _handle_request_message(&mut self, umsg: UMessage) {
+
+        let rt_ = Runtime::new().unwrap();
+        if let Some(listner_array) = self
+            .listner_map
+            .lock()
+            .unwrap()
+            .get(&umsg.attributes.sink.to_string())
+        {
+            for listner_ref in listner_array {
+                rt_.block_on(async {
+                let _ = listner_ref.on_receive(umsg.clone()).await;
+            });
+        }
+    }
+}
+}
+#[async_trait]
+impl UTransport for UtransportSocket {
+    /// Sends a message using this transport's message exchange mechanism.
+    ///
+    /// # Arguments
+    ///
+    /// * `message` - The message to send. The `type`, `source` and`sink` properties of the [`crate::UAttributes`] contained
+    ///   in the message determine the addressing semantics:
+    ///   * `source` - The origin of the message being sent. The address must be resolved. The semantics of the address
+    ///     depends on the value of the given [attributes' type](crate::UAttributes::type_) property .
+    ///     * For a [`PUBLISH`](crate::UMessageType::UMESSAGE_TYPE_PUBLISH) message, this is the topic that the message should be published to,
+    ///     * for a [`REQUEST`](crate::UMessageType::UMESSAGE_TYPE_REQUEST) message, this is the *reply-to* address that the sender expects to receive the response at, and
+    ///     * for a [`RESPONSE`](crate::UMessageType::UMESSAGE_TYPE_RESPONSE) message, this identifies the method that has been invoked.
+    ///   * `sink` - For a `notification`, an RPC `request` or RPC `response` message, the (resolved) address that the message
+    ///     should be sent to.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the message could not be sent.
+    async fn send(&self, message: UMessage) -> Result<(), UStatus> {
+        let mut socket_clone = self
+            .socket_sync
+            .try_clone()
+            .expect("dispatcher socket connection cloning failed");
+
+        let umsg_serialized = message.clone().write_to_bytes().expect("Send Serialization Issue");
+        let _ = UMessage::parse_from_bytes(&umsg_serialized.clone()).expect("Failed to parse message");
```

Comment:
Recommend handling the `Result`. But in this case since you used `.expect()` nothing is returned anyway so `let _ =` is not needed.